### ISSUE TITLE
是否可以只生成前视的视频

### ISSUE THREAD

**Issue by wofeifeileni** [SenseTime-FVG/OpenDWM#62]
是否可以只生成前视的视频，像主页视频demo2_3_out2.mp4开头展示的那种

![Image](https://github.com/user-attachments/assets/d378329e-3a77-424d-a6b5-cf819d66b4da)

**Comment by johnren-code** [comment]
你好，我刚才基于`ctsd_unimlvg_6views_video_generation.json`进行魔改，得到了ctsd_unimlvg_single_views_video_generation.json:
```
{
    "device": "cuda",
    "generator_seed": 0,
    "pipeline": {
        "_class_name": "dwm.pipelines.ctsd.CrossviewTemporalSD",
        "common_config": {
            "frame_prediction_style": "ctsd",
            "cat_condition": true,
            "cond_with_action": false,
            "condition_on_all_frames": true,
            "explicit_view_modeling": false,
            "disable_crossview": true,
            "autocast": {
                "device_type": "cuda"
            },
            "text_encoder_load_args": {
                "torch_dtype": {
                    "_class_name": "get_class",
                    "class_name": "torch.float16"
                },
                "quantization_config": {
                    "_class_name": "diffusers.quantizers.quantization_config.BitsAndBytesConfig",
                    "load_in_4bit": true,
                    "bnb_4bit_quant_type": "nf4",
                    "bnb_4bit_compute_dtype": {
                        "_class_name": "get_class",
                        "class_name": "torch.float16"
                    }
                }
            },
            "memory_efficient_batch": 12
        },
        "training_config": {},
        "inference_config": {
            "guidance_scale": 4,
            "inference_steps": 40,
            "preview_image_size": [
                448,
                252
            ],
            "sequence_length_per_iteration": 19,
            "reference_frame_count": 1,
            "generate_frames_for_reference": false,
            "autoregression_data_exception_for_take_sequence": [
                "crossview_mask"
            ]
        },
        "model": {
            "_class_name": "dwm.models.crossview_temporal_dit.DiTCrossviewTemporalConditionModel",
            "attention_head_dim": 64,
            "caption_projection_dim": 1536,
            "in_channels": 16,
            "joint_attention_dim": 4096,
            "num_attention_heads": 24,
            "num_layers": 24,
            "out_channels": 16,
            "patch_size": 2,
            "pooled_projection_dim": 2048,
            "pos_embed_max_size": 192,
            "sample_size": 128,
            "perspective_modeling_type": "",
            "enable_crossview": false,
            "crossview_attention_type": "rowwise",
            "crossview_block_layers": [
                1,
                5,
                9,
                13,
                17,
                21
            ],
            "crossview_gradient_checkpointing": true,
            "enable_temporal": true,
            "temporal_attention_type": "full",
            "temporal_block_layers": [
                1,
                5,
                9,
                13,
                17,
                21
            ],
            "temporal_gradient_checkpointing": true,
            "mixer_type": "AlphaBlender",
            "merge_factor": 2,
            "condition_image_adapter_config": {
                "in_channels": 6,
                "channels": [
                    1536,
                    1536,
                    1536,
                    1536,
                    1536,
                    1536
                ],
                "is_downblocks": [
                    true,
                    false,
                    false,
                    false,
                    false,
                    false
                ],
                "num_res_blocks": 2,
                "downscale_factor": 8,
                "use_zero_convs": true
            }
        },
        "model_dtype": {
            "_class_name": "get_class",
            "class_name": "torch.float16"
        },
        "pretrained_model_name_or_path": "/Data/rbl/project/OpenDWM/pretrained_models/stable-diffusion-3-medium-diffusers",
        "model_checkpoint_path": "/Data/rbl/project/OpenDWM/checkpoints/ctsd_unimlvg_tirda_bm_nwa_60k.pth",
        "model_load_state_args": {
            "strict": false
        }
    },
    "validation_dataset": {
        "_class_name": "dwm.datasets.common.DatasetAdapter",
        "base_dataset": {
            "_class_name": "dwm.datasets.preview.PreviewDataset",
            "json_file": "/Data/rbl/project/OpenDWM/resources/front_view_unstructure/data.json",
            "sequence_length": 179,
            "fps_stride_tuples": [
                [
                    10,
                    60
                ]
            ],
            "sensor_channels": [
                "CAM_FRONT"
            ],                        
            "enable_camera_transforms": false,
            "use_hdmap": false,
            "use_3dbox": false,
            "drop_vehicle_color": true,
            "stub_key_data_dict": {
                "crossview_mask": [
                    "content",
                    {
                        "_class_name": "torch.tensor",
                        "data": {
                            "_class_name": "json.loads",
                            "s": "[[1,1,0,0,0,1],[1,1,1,0,0,0],[0,1,1,1,0,0],[0,0,1,1,1,0],[0,0,0,1,1,1],[1,0,0,0,1,1]]"
                        },
                        "dtype": {
                            "_class_name": "get_class",
                            "class_name": "torch.bool"
                        }
                    }
                ]
            }
        },
        "transform_list": [
            {
                "old_key": "images",
                "new_key": "vae_images",
                "transform": {
                    "_class_name": "torchvision.transforms.Compose",
                    "transforms": [
                        {
                            "_class_name": "torchvision.transforms.Resize",
                            "size": [
                                256,
                                448
                            ]
                        },
                        {
                            "_class_name": "torchvision.transforms.ToTensor"
                        }
                    ]
                }
            },
            {
                "old_key": "image_description",
                "new_key": "clip_text",
                "transform": {
                    "_class_name": "dwm.datasets.common.Copy"
                },
                "stack": false
            }
        ],
        "pop_list": [
            "images",
            "image_description"
        ]
    },
    "validation_dataloader": {
        "batch_size": 1,
        "num_workers": 1,
        "collate_fn": {
            "_class_name": "dwm.datasets.common.CollateFnIgnoring",
            "keys": [
                "clip_text"
            ]
        }
    },
    "preview_dataloader": {
        "batch_size": 1,
        "num_workers": 1,
        "collate_fn": {
            "_class_name": "dwm.datasets.common.CollateFnIgnoring",
            "keys": [
                "clip_text"
            ]
        }
    }
}
```
和自定义的一个data.json，我设置参考帧数量为1，然后只给data.json的第0个元素一个初始的rgb值，其他全部置为null，在代码配置中，我把bbox，map和camera transform相关的condition全关了，并且在推理的时候忽略了cross-view block，我想复现一下基于Vista的那种输入第一帧，进行演化的结果，但是我得到的结果非常糟糕，这是为什么呢？

https://github.com/user-attachments/assets/45440ee3-0b2c-4cd0-a88c-aabc78bd0d9f

**Comment by johnren-code** [comment]
您好，我想问一下您，`mini_batch`的具体含义是什么呢，以及为什么在OpenDV数据集上训练的时候设置`cross_view=True`呢

**Comment by wzhgba** [comment]
条件差异太大了，最好根据实际使用条件的情况再训练一下。
mini_batch 主要用于在 OpenDV 数据集中将多个样本的片段拼接在视角维度上，来支持和其他多视角数据集的混合训练，不过注意这种情况下 OpenDV 数据集上需要把 crossview mask 设置为 torch.eye(mini_batch) 矩阵。

**Comment by wzhgba** [comment]
是，模型适用的 inference 规格应以最后阶段的训练规格为准，UniMLVG 模型的最后阶段是在多视角任务上训练的，且没有跨视角模块的 drop 机制，因此模型会过拟合于多视角的输入。要同时支持多视角和单视角任务，需要在最终训练阶段同时包含这两类任务。

**Comment by johnren-code** [comment]
你好，那你们在演示视频中放的跟Vista对比的单视角视频生成结果，不是使用第三阶段的模型权重推理得到的吗？是第一阶段的吗
